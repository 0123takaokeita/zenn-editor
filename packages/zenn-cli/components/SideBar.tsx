import Link from "next/link";
import { useRouter } from "next/router";
import { NavCollections, NavCollection, NavItem } from "@types";

const SideBarNavItem: React.FC<{ navItem: NavItem }> = ({ navItem }) => {
  const router = useRouter();
  const className = `sidebar-nav-item ${
    router.asPath === navItem.realPath ? "active" : ""
  }`;
  return (
    <Link
      href={navItem.dynamicRoutePath || navItem.realPath}
      as={navItem.realPath}
    >
      <a
        href={navItem.realPath}
        className={className}
        title={`${navItem.realPath}.md`}
      >
        {navItem.name}
      </a>
    </Link>
  );
};

const SideBarNavCollection: React.FC<{ navCollection: NavCollection }> = ({
  navCollection,
}) => {
  return (
    <details className="sidebar-collection" open>
      <summary className="sidebar-collection__title">
        {navCollection.name}
      </summary>
      <ul className="sidebar-collection__ul">
        {navCollection?.items.map((navItem) => (
          <SideBarNavItem navItem={navItem} key={navItem.realPath} />
        ))}
      </ul>
    </details>
  );
};

const LogoLink: React.FC = () => (
  <Link href="/">
    <a href="/" className="sidebar-logo-link">
      <img src="/logo.svg" alt="Zenn Editor" width={187} height={22} />
    </a>
  </Link>
);

const SideBar: React.FC<{
  navCollections: NavCollections;
}> = ({ navCollections }) => {
  return (
    <aside className="sidebar">
      <LogoLink />
      <div className="sidebar-collections">
        {navCollections.map((navCollection) => (
          <SideBarNavCollection
            navCollection={navCollection}
            key={navCollection.name}
          />
        ))}
      </div>
      <a
        href="https://zenn.dev/dashboard/contents/uploader"
        className="sidebar-external-link"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src="https://twemoji.maxcdn.com/2/svg/\1f4f7.svg"
          width="16"
          height="16"
        />
        画像のアップロード
      </a>
    </aside>
  );
};

export default SideBar;
