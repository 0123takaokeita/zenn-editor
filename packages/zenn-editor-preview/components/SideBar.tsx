import Link from "next/link";
import { NavCollections, NavCollection, NavItem } from "@types";

const SideBarNavItem: React.FC<{ navItem: NavItem }> = ({ navItem }) => {
  return (
    <Link
      href={navItem.dynamicRoutePath || navItem.realPath}
      as={navItem.realPath}
    >
      <a href={navItem.realPath} className="sidebar-nav-item">
        {navItem.name}
      </a>
    </Link>
  );
};

const SideBarNavCollection: React.FC<{ navCollection: NavCollection }> = ({
  navCollection,
}) => {
  return (
    <details className="sidebar-collection" open>
      <summary className="sidebar-collection__title">
        {navCollection.name}
      </summary>
      <ul className="sidebar-collection__ul">
        {navCollection?.items.map((navItem) => (
          <SideBarNavItem navItem={navItem} key={navItem.realPath} />
        ))}
      </ul>
    </details>
  );
};

const SideBar: React.FC<{
  navCollections: NavCollections;
}> = ({ navCollections }) => {
  return (
    <aside className="sidebar">
      {navCollections.map((navCollection) => (
        <SideBarNavCollection
          navCollection={navCollection}
          key={navCollection.name}
        />
      ))}
    </aside>
  );
};

export default SideBar;
