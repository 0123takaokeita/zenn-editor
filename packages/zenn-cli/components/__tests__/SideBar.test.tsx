import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import { SideBar } from "@components/SideBar";
import { NavCollection, NavCollections } from "@types";

const renderer = createRenderer();
const dummyNavigation: NavCollection = {
  items: [{ name: "", href: "" }],
  name: "",
};
const dummyNavigations: NavCollections = [dummyNavigation];

describe("<SideBar />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<SideBar navCollections={dummyNavigations} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
