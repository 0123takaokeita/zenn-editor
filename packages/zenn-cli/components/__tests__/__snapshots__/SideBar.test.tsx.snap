// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render and match the snapshot 1`] = `
<aside
  className="sidebar"
>
  <SidebarHeader />
  <div
    className="sidebar-collections"
  >
    <SideBarNavCollection
      navCollection={
        Object {
          "items": Array [
            Object {
              "href": "",
              "name": "test-navigation",
            },
          ],
          "name": "test-navigations",
        }
      }
    />
  </div>
  <a
    className="sidebar-external-link"
    href="https://zenn.dev/dashboard/uploader"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      height="16"
      src="https://twemoji.maxcdn.com/2/svg/\\\\1f4f7.svg"
      width="16"
    />
    画像のアップロード
  </a>
  <a
    className="sidebar-external-link"
    href="https://zenn.dev/zenn/articles/markdown-guide"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      height="16"
      src="https://twemoji.maxcdn.com/2/svg/\\\\1f58b.svg"
      width="16"
    />
    マークダウン・ガイド
  </a>
  <a
    className="sidebar-external-link"
    href="https://zenn.dev/zenn/articles/zenn-cli-guide"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      height="16"
      src="https://twemoji.maxcdn.com/2/svg/\\\\1f4bb.svg"
      width="16"
    />
    CLIコマンド・ガイド
  </a>
</aside>
`;
